client/verta/verta/registry/entities/_modelversion.py

```diff
@@ -1075,3 +1093,297 @@ def delete(self):
         endpoint = "/api/v1/registry/model_versions/{}".format(self.id)
         response = self._conn.make_proto_request("DELETE", endpoint)
         self._conn.must_response(response)
+
+    @staticmethod
+    def _get_metadata_for_df(df):
+        """Capture info about `df`'s cols to determine how to profile them.
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            DataFrame to be profiled downstream.
+
+        Returns
+        -------
+        dict
+            ``"num_unique"`` and ``"type"`` for each column in `df`.
+
+        """
+        metadata = {}
+        for column in df:
+            metadata[column] = {}
+            metadata[column]["num_unique"] = df[column].value_counts().size
+            metadata[column]["type"] = str(df[column].dtypes)
+
+        return metadata
```

Comment:
```suggestion
        return {column: metadata for column, metadata in df.apply(lambda col: {"num_unique": col.value_counts().size, "type": str(col.dtypes)}
```

Reply:
(Or something similar using `df.apply`)

Reply:
Done! I do remember reading that `df.apply()` is much more efficient than `for`-looping over the DataFrame.